skimage/draw/_random_shapes.py

```diff
@@ -200,13 +205,34 @@ def _generate_random_colors(num_colors, num_channels, intensity_range, random):
         each channel are drawn from the corresponding `intensity_range`.
 
     """
+
     if num_channels == 1:
-        intensity_range = (intensity_range, )
+        intensity_range_tpl = (intensity_range, )
     elif len(intensity_range) == 1:
-        intensity_range = intensity_range * num_channels
+        intensity_range_tpl = intensity_range * num_channels
+    else:
+        intensity_range_tpl = intensity_range
+
+    if np.diff(np.array(intensity_range_tpl)).sum() == 0:
```

Comment:
Is this check overkill?  Even if not, I think the error message should at least be improved.  But what do we lose if we remove it entirely?

Reply:
If intensity range spans only a single color, and this color equals the color to be excluded, the recursive call in line 230 will end up in an infinite loop. Therefore this check is important. The error message could be more specific, indeed.